frontend/src/components/InfiniteContentList.js

```diff
@@ -0,0 +1,64 @@
+import Component from '../core/Component';
+import ContentCard from './ContentCard';
+import EmptyMessage from './EmptyMessage';
+
+const DATA_LIMIT = 40;
+
+export default class InfiniteContentList extends Component {
+  setup() {
+    const { category, type, initialData, onLastItemShown } = this.$props;
+    this.$state = {
+      category,
+      data: initialData,
+      type,
+      $emptyMessage: null,
+    };
+    this.$io = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.intersectionRatio > 0 && entry.target.dataset.index < DATA_LIMIT) {
+          if (entry.target.dataset.index == this.$state.data.length) {
+            onLastItemShown(category);
+          }
+        }
+      });
+    });
+  }
+
+  template() {
+    return `
+      <h2 class="content-section-title">${this.$state.category}</h2>
+      <section class="content-list" data-component="content-list"></section>
+    `;
+  }
+
+  mounted() {
+    const $list = this.$target.querySelector('[data-component="content-list"]');
+    const { data, type } = this.$state;
+    const { handleCardClick, handleBookmark } = this.$props;
+
+    const hasData = data && data.length > 0 ? true : false;
+
+    if (!hasData) {
+      this.$state.$emptyMessage = new EmptyMessage($list, { message: 'No Contents :(', visible: !hasData });
+      this.$state.$emptyMessage.Visible = !hasData;
+    } else {
```

Comment:
```suggestion
    if (!hasData) {
      this.$state.$emptyMessage = new EmptyMessage($list, { message: 'No Contents :(', visible: !hasData });
      this.$state.$emptyMessage.Visible = !hasData;
      return;
    }
```
이렇게 else로 진입하는 블록을 없애는 방식으로 작성해주세요! 
코드의 block 이 깊어질수록 나중에 이해하기가 힘들어진답니다~